quint/src/cliCommands.ts

```diff
@@ -604,6 +606,67 @@ export async function runSimulator(prev: TypecheckedStage): Promise<CLIProcedure
   }
 }
 
+/** compile a spec and output in the specified `typechecked.args.target`
+ *
+ * The output produced by this command includes the parsed, typechecked, and
+ * flattened data.
+ *
+ * @param typechecked The result of a preceding typechecking stage
+ */
+// TODO This should share most of the logic in `verifySpec`, but the logic in
+// that command would need some significant reworking to make it composable
```

Comment:
Perhaps, whatever we decide to do on https://github.com/informalsystems/quint/pull/1399/files#r1521344080 might help with this as well.

Reply:
Yeah, I think that is exactly right! Most of complication in `verifySpec` is only required due to the flattening problem.